.github/workflows/ci.yaml

```diff
@@ -368,116 +336,363 @@ jobs:
+          echo "SDKROOT=$(xcrun -sdk macosx --show-sdk-path)" >> $GITHUB_ENV
+          echo "MACOSX_DEPLOYMENT_TARGET=$(xcrun -sdk macosx --show-sdk-platform-version)" >> $GITHUB_ENV
+
+      - name: Configure artifact names (libc)
+        if: ${{ matrix.libc }}
+        shell: bash
+        run: |
+          echo "SHORT_TARGET_NAME=${{ matrix.platform }}-${{ matrix.architecture }}-${{ matrix.libc }}" >> $GITHUB_ENV
+          echo "PRE_GYP_TARGET_NAME=${{ matrix.platform }}-${{ matrix.architecture }}-${{ matrix.libc }}" >> $GITHUB_ENV
+
+      - name: Configure artifact names (not libc)
+        if: ${{ ! matrix.libc }}
+        shell: bash
+        run: |
+          echo "SHORT_TARGET_NAME=${{ matrix.platform }}-${{ matrix.architecture }}" >> $GITHUB_ENV
+          echo "PRE_GYP_TARGET_NAME=${{ matrix.platform }}-${{ matrix.architecture }}-unknown" >> $GITHUB_ENV
+
+      - name: Build - Node
+        if: matrix.target != 'x86_64-unknown-linux-musl'
+        working-directory: node-bindings
+        run: |
+          npm install --ignore-scripts
+          npm run build-${{ env.SHORT_TARGET_NAME }}
+
+      - name: Build - Node (linux-musl)
+        if: matrix.target == 'x86_64-unknown-linux-musl'
+        uses: docker://rust:alpine3.15
+        env:
+          RUSTFLAGS: -C target-feature=-crt-static
+        with:
+          entrypoint: /bin/sh
+          args: -c "
+            cd node-bindings &&
+            apk add alpine-sdk nodejs npm git &&
+            npm install --ignore-scripts &&
+            npm run build-${{ env.SHORT_TARGET_NAME }}"
+
+      - name: Compress node artifact
+        shell: bash
+        run: tar -C node-bindings -zcvf stacks-devnet-js-${{ env.PRE_GYP_TARGET_NAME }}.tar.gz native/index.node
+
+      - name: Upload node artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: stacks-devnet-js-${{ env.PRE_GYP_TARGET_NAME }}
+          path: stacks-devnet-js-${{ env.PRE_GYP_TARGET_NAME }}.tar.gz
+
+      # - name: Unit Tests - Node
+      #   working-directory: node-bindings
+      #   run: npm run spec
+
       - name: Upload Node Artifact to GH release
         uses: svenstaro/upload-release-action@v2
-        if: startsWith(github.ref, 'refs/tags/v')
+        if: startsWith(github.ref, 'refs/heads/main') && needs.get_release_info.outputs.tag != ''
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
           file: stacks-devnet-js-${{ env.PRE_GYP_TARGET_NAME }}.tar.gz
           tag: ${{ github.ref }}
 
+      - name: NPM Publish
+        uses: JS-DevTools/npm-publish@v1
+        if: matrix.target == 'x86_64-unknown-linux-gnu' && startsWith(github.ref, 'refs/heads/main') && needs.get_release_info.outputs.tag != ''
+        with:
+          package: node-bindings/package.json
+          access: public
+          greater-version-only: true
+          token: ${{ secrets.NPM_TOKEN }}
+
+  dist_oreo:
+    name: Build Oreo Distributions
+    runs-on: ${{ matrix.os }}
```

Comment:
I'd be perfectly fine with the idea of just considering unix for Oreo (instead of currently macOS/windows/unix - if my reading is correct)

Reply:
I only put down one dist variant for Oreo because I wasn't sure if you wanted to create more like we do for Clarinet. But if you only need one variant, I'll simplify this section a bit.